avm/res/azure-stack-hci/cluster/main.bicep

```
@@ -0,0 +1,402 @@
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource cluster 'Microsoft.AzureStackHCI/clusters@2024-04-01' = {
+  name: name
+  identity: {
+    type: 'SystemAssigned'
+  }
+  location: location
+  properties: {}
+  tags: tags
+}
+
+@batchSize(1)
+module deploymentSetting 'deployment-settings/main.bicep' = [
+  for deploymentOperation in sortedDeploymentOperations: {
+    name: 'deploymentSettings-${deploymentOperation}'
+    params: {
+      cloudId: cluster.properties.cloudId
+      clusterName: name
+      clusterNodeNames: clusterNodeNames
+      clusterWitnessStorageAccountName: clusterWitnessStorageAccountName
+      customLocationName: customLocationName
+      defaultGateway: defaultGateway
+      deploymentMode: deploymentOperation
+      deploymentPrefix: deploymentPrefix
+      dnsServers: dnsServers
+      domainFqdn: domainFqdn
+      domainOUPath: domainOUPath
+      enableStorageAutoIp: enableStorageAutoIp
+      endingIPAddress: endingIPAddress
+      episodicDataUpload: episodicDataUpload
+      isEuropeanUnionLocation: isEuropeanUnionLocation
+      keyVaultName: keyVaultName
+      networkIntents: networkIntents
+      securityConfiguration: securityConfiguration
+      startingIPAddress: startingIPAddress
+      storageConfigurationMode: storageConfigurationMode
+      storageConnectivitySwitchless: storageConnectivitySwitchless
+      storageNetworks: storageNetworks
+      streamingDataClient: streamingDataClient
+      subnetMask: subnetMask
+    }
+  }
+]
+
+resource cluster_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(cluster.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: cluster
+  }
+]
+
+@description('The name of the cluster.')
+output name string = cluster.name
+@description('The ID of the cluster.')
+output resourceId string = cluster.id
+@description('The resource group of the cluster.')
+output resourceGroupName string = resourceGroup().name
+@description('The managed identity of the cluster.')
+output systemAssignedMIPrincipalId string = cluster.identity.principalId
+@description('The location of the cluster.')
+output location string = cluster.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+@export()
+type networkIntent = {
```

```suggestion
type networkIntentType = {
```
Out of consistency, it might be a good call to use the `Type` suffix for user defined types.